src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -71,7 +71,12 @@ JobExecutorType(MultiPlan *multiPlan)
 												 " queries on the workers.")));
 	}
 
-	workerNodeList = ActivePrimaryNodeList();
+	if (multiPlan->operation != CMD_SELECT)
+	{
+		ereport(FATAL, (errmsg("we assume only SELECT queries hit this point")));
```

Comment:
I don't think that's true though, I think `JobExecutorType` is always called, I just want to be able to `Assert` that by the time control reaches here we're only playing with `SELECT`.